instrumentation/pekko/pekko-actor-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pekkoactor/v1_0/ExecutionContextWrapper.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pekkoactor.v1_0;
+
+import io.opentelemetry.context.Context;
+import scala.concurrent.ExecutionContext;
+
+public class ExecutionContextWrapper implements ExecutionContext {
+  private final ExecutionContext executionContext;
+  private final Context otelContext;
+
+  public ExecutionContextWrapper(ExecutionContext executionContext, Context otelContext) {
+    this.executionContext = executionContext;
+    this.otelContext = otelContext;
+  }
+
+  @Override
+  public void execute(Runnable runnable) {
+    executionContext.execute(otelContext.wrap(runnable));
+  }
+
+  @Override
+  public void reportFailure(Throwable cause) {
+    executionContext.reportFailure(cause);
```

Comment:
if user code can observe this through a callback then it might make sense to restore the otel context around `reportFailure` call.

Reply:
I've removed this code